packages/docusaurus-theme-classic/src/theme/Heading/index.tsx

```diff
@@ -18,10 +18,8 @@ import styles from './styles.module.css';
 const Heading = (Tag: HeadingType): ((props: Props) => JSX.Element) =>
   function TargetComponent({id, ...props}) {
     const {
-      siteConfig: {
-        themeConfig: {navbar: {hideOnScroll = false} = {}} = {},
-      } = {},
-    } = useDocusaurusContext();
+      hideOnScroll = false,
+    } = useDocusaurusContext().siteConfig.themeConfig.navbar;
```

Comment:
navbar can be null (see Joi validation, there is no .require() call nor .default() so it is optional)